src/node_binding.cc

```diff
@@ -619,7 +612,20 @@ void GetLinkedBinding(const FunctionCallbackInfo<Value>& args) {
   Local<String> module_name = args[0].As<String>();
 
   node::Utf8Value module_name_v(env->isolate(), module_name);
-  node_module* mod = get_linked_module(*module_name_v);
+  const char* name = *module_name_v;
+  node_module* mod = nullptr;
+
+  // Iterate from here to the nearest non-Worker Environment to see if there's
+  // a linked binding defined locally rather than through the global list.
+  Environment* cur_env = env;
+  while (mod == nullptr && cur_env != nullptr) {
+    Mutex::ScopedLock lock(cur_env->extra_linked_bindings_mutex());
+    mod = FindModule(cur_env->extra_linked_bindings_head(), name, NM_F_LINKED);
+    cur_env = cur_env->worker_parent_env();
+  }
```

Comment:
@addaleax - does this mean:
 - a binding that is already available in any of the workers in the ancestor chain will cause this env not to load it again
 - and this worker will be able to use that binding from the parent / ancestor worker?

Reply:
@gireeshpunathil The module registration function will be called for each Environment/Worker … does that answer your question?

Reply:
sorry, but not really. IIUC, after this loop exits, the effect is that `mod` will point to one of the ancestor environment, if available, and later that will be used to link with `this` (current) env / worker. Is that the intent? (I should admit that I do not completely follow the module registration logic)

Reply:
@gireeshpunathil Tbh, I don’t quite understand what you’re saying… can you clarify what you mean by “`mod` will point to” or “will be used to link with”?

`mod` doesn’t point to any `Environment` itself. It is stored on an `Environment` instance if it is added through the new API here, meaning that it can only be used while that `Environment` is alive (that’s what the doc comment in `node.h` aims to make clear).

When `process._linkedBinding()` is actually called in that `Environment` or one of its Workers, that `mod` entry is looked up, but it is not used afterwards in the same `Environment`; in particular, the created addon does not refer back to the `node_module` struct that was used to create it.

Does that help?

Reply:
thanks for the clarification, I am good with that explanation!